File: projects/go-sdk/handlers/unseal.go
package handlers

import (
	"encoding/json"
	"go-sdk/response"
	"net/http"

	"github.com/fingerprintjs/fingerprint-pro-server-api-go-sdk/v7/sdk/sealed"
)

type RequestBody struct {
	SealedData []byte                 `json:"sealedData"`
	Keys       []sealed.DecryptionKey `json:"keys"`
}

func Unseal(w http.ResponseWriter, r *http.Request) {
	var body RequestBody
	decoder := json.NewDecoder(r.Body)
	decoder.Decode(&body)

	unsealedResponse, err := sealed.UnsealEventsResponse(body.SealedData, body.Keys)

	var resp response.MusicianResponse
	if err != nil {
		resp = response.MusicianResponse{
			Code:             500,
			OriginalResponse: "",
			ParsedResponse:   err.Error(),
		}
	} else {
		resp = response.MusicianResponse{
			Code:             200,
			OriginalResponse: "",
			ParsedResponse:   unsealedResponse,
		}
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(200)
	json.NewEncoder(w).Encode(resp)
}

File: projects/node-sdk/src/handlers/unseal.ts
import { Handler, MusicianResponse } from '../types'
import {
  EventsGetResponse,
  unsealEventsResponse,
  DecryptionKey,
  DecryptionAlgorithm,
} from '@fingerprintjs/fingerprintjs-pro-server-api'
import { unwrapError } from '../utils'

interface RequestBody {
  sealedData: string
  keys: { key: string; algorithm: string }[]
}

export const unseal: Handler<{}, RequestBody> = async (req, res) => {
  const { sealedData, keys } = req.body

  const sealedDataBuffer = Buffer.from(sealedData, 'base64')
  const unsealKeys: DecryptionKey[] = keys.map(({ key, algorithm }) => ({
    key: Buffer.from(key, 'base64'),
    algorithm: algorithm as DecryptionAlgorithm,
  }))

  let result: MusicianResponse<EventsGetResponse>
  try {
    const event = await unsealEventsResponse(sealedDataBuffer, unsealKeys)
    result = {
      code: 200,
      originalResponse: event,
      parsedResponse: event,
    }
  } catch (error) {
    result = await unwrapError<EventsGetResponse>(error)
  }
  res.send(result)
}
